packaging/libaec-config.cmake

```
@@ -0,0 +1,74 @@
+# libaec-config.cmake
+# -------------------
+#
+# Finds the AEC library, specify the starting search path in libaec_ROOT.
+#
+# Static vs. shared
+# -----------------
+# To make use of the static library instead of the shared one, one
+# needs to set the variable libaec_USE_STATIC_LIBS to ON before
+# calling find_package.
+# Example:
+#   set(libaec_USE_STATIC_LIBS ON)
+#   find_package(libaec CONFIG)
+#
+# This will define the following variables:
+#
+#   libaec_FOUND   - True if the system has the AEC library.
+#   libaec_VERSION - The version of the AEC library which was found.
+#   SZIP_FOUND     - True if the system has the SZIP library.
+#   SZIP_VERSION   - The version of the SZIP library which was found.
+#   SZIP_LIBRARIES - All the required libraries to make use of SZIP.
+#
+# and the following imported targets:
+#
+#   libaec::aec-shared  - The shared AEC library target (if it was built).
+#   libaec::sz-shared   - The shared SZIP compatible version of the AEC library
+#                         (if it was built).
+#   libaec::aec-static  - The static AEC library target (if it was built).
+#   libaec::sz-static   - The static SZIP compatible version of the AEC library
+#                         (if it was built).
+#   libaec::aec         - The (shared or static) AEC library target (according
+#                         to the value of libaec_USE_STATIC_LIBS).
+#   libaec::sz          - The (shared or static) SZIP compatible version of the
+#                         AEC library (according to the value of
+#                         libaec_USE_STATIC_LIBS).
+
+include(${CMAKE_CURRENT_LIST_DIR}/libaec-targets.cmake)
+
+# Support legacy variables
+find_path(SZIP_INCLUDE_DIR NAMES szlib.h DOC "SZIP include directory")
+if (MSVC)
+  find_library(SZIP_STATIC
+    NAMES szip-static.lib DOC "SZIP compatible version of the AEC library")
+else ()
+  find_library(SZIP_STATIC
+    NAMES libsz.a DOC "SZIP compatible version of the AEC library")
+endif ()
```

Similar to above. Also untested:
``` suggestion
if (TARGET libaec::sz-static)
  get_target_property(SZIP_STATIC libaec::sz-static IMPORTED_IMPLIB)
  if (NOT SZIP_STATIC)
    get_target_property(_lib_check libaec::sz-static IMPORTED_LOCATION)
    if (EXISTS ${_lib_check})
      set(SZIP_STATIC ${_lib_check})
    endif ()
  endif ()
endif ()
```

---

Trying to get the `IMPORTED_IMPLIB` property is probably only needed for shared libraries.

---

On Linux, `IMPORTED_LOCATION` wasn't found. `LOCATION`  worked. Have to check on Windows.

---

Ah. I'm remembering now. The variables are suffixed with an identifier for the build type (`CMAKE_BUILD_TYPE`).

Re-reading the CMake documentation, it would probably be best to first query [IMPORTED_IMPLIB](https://cmake.org/cmake/help/v4.0/prop_tgt/IMPORTED_IMPLIB.html#prop_tgt:IMPORTED_IMPLIB) (for platforms like Windows, AIX or Apple) and, if that is empty, query [LOCATION](https://cmake.org/cmake/help/v4.0/prop_tgt/LOCATION.html#prop_tgt:LOCATION). 
(See also the last two sentences in [IMPORTED_LOCATION](https://cmake.org/cmake/help/v4.0/prop_tgt/IMPORTED_LOCATION.html#prop_tgt:IMPORTED_LOCATION).)

---

`IMPORTED_IMPLIB` seems not to be set. At least not for multi-configuration generators like Visual Studio. What works here is something like
```suggestion
if (TARGET libaec::sz)
  get_target_property(_szip_config libaec::sz IMPORTED_CONFIGURATIONS)
  if (_szip_config)
    list(GET _szip_config 0 _szip_config)
  endif ()
  get_target_property(SZIP_LIBRARIES
    libaec::sz "IMPORTED_IMPLIB_${_szip_config}")
  if (NOT SZIP_LIBRARIES)
    get_target_property(_lib_check
      libaec::sz "IMPORTED_LOCATION_${_szip_config}")
    if (EXISTS ${_lib_check})
      set(SZIP_LIBRARIES ${_lib_check})
    endif ()
  endif ()
endif ()
```
But I'm unsure if just taking the first configuration is the correct way to do it.

---

Ah. All of that is coming back to me now.
I proposed some changes for SuiteSparse in the past that were dealing with this. See, e.g., https://github.com/DrTimothyAldenDavis/SuiteSparse/blob/9f0eedfa780203ed462faede7f51ee250e12c64a/AMD/Config/AMDConfig.cmake.in#L82-L136.

I prefer your `get_target_property(_szip_config libaec::sz IMPORTED_CONFIGURATIONS)` compared to the hardcoded build type identifiers there. 
But like you already wrote, it might be necessary to iterate over all imported configurations (ideally starting with the current build type used in the project that is importing the targets - if it isn't a Multi-Config CMake generator) and check for which files actually exist. (I seem to recall that that was necessary there.)

That general approach was originally implemented in SuiteSparse approximately two years ago. And even if it doesn't cover all imaginable situations, there haven't been reports about issues since then as far as I can tell.

---

Okay, then I'll take that approach. It's probably never going to be perfect. Especially for multi-config generators.